amundsen_application/static/js/ducks/tableMetadata/types.ts

```diff
@@ -0,0 +1,171 @@
+import { PreviewData, PreviewQueryParams, TableMetadata, User } from '../../components/TableDetail/types';
+import { UpdateTagData, Tag } from '../../components/Tags/types';
+import { UpdateMethod } from '../../components/OwnerEditor/types';
+export { PreviewData, PreviewQueryParams, TableMetadata, Tag, User, UpdateMethod, UpdateTagData };
+
+export type DescriptionResponse = { description: string; } & MessageResponse;
+export type LastIndexedResponse = { timestamp: string; } & MessageResponse;
+export type MessageResponse = { msg: string };
+export type PreviewDataResponse = { previewData: PreviewData; } & MessageResponse;
+export type TableDataResponse = { tableData: TableMetadata & { owners: User[], tags: Tag[] } } & MessageResponse;
+
```

Comment:
Seems like the more complex data types could benefit from an explicit type definition. Using a sequence of `&`s makes this difficult to read.

At the very least I think `MessageResponse` should go on top since others depend on it.

Reply:
Difficult to read but I think building related types this way makes it easier to implement changes and additions.

Granted, l will actually define a type for the tableData of the `TableDataResponse` before merging.  That one is unnecessarily hard to read -- I was curious if the syntax `TableMetadata & { owners: User[], tags: Tag[] }` would work and forgot to clean up.